File: 53.maximum-subarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = nums[0]
        sum = 0
        for num in nums:
            sum = max(sum, 0)
            sum += num
            ans = max(ans, sum)
        return ans

File: 53.maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let ans = nums[0];
    let sum = nums[0];
    for (let i = 1; i < nums.length; i++) {
        sum = Math.max(sum, 0);
        sum += nums[i];
        ans = Math.max(ans, sum);
    }
    return ans;
};